### PR TITLE
Port swift-demangle to Windows

### PR DESCRIPTION
**Pull request by hughbe**
Windows doesn't support `unistd.h`. Therefore, it doesn't expose any kind of `getline` posix function.

Instead, let's try to make the swift-demangle.cpp file platform independent by using `std::getline`

I've marked this as WIP as I'm not sure how similar `getline` and `std::getline` are. I'm also getting failures running the swift-demangle unit tests, but wanted to get this out there to see what I'm doing wrong!

### CODE REVIEW

**Comment by hughbe** [comment]
@swift-ci please smoke test

File: tools/swift-demangle/swift-demangle.cpp
Comment (compnerd) [inline tools/swift-demangle/swift-demangle.cpp]: I think that I would rather see this as

    #if defined (__unix__) || (defined(__APPLE__) && defined(__MACH__))

File: tools/swift-demangle/swift-demangle.cpp
Comment (compnerd) [inline tools/swift-demangle/swift-demangle.cpp]: What is this header pulling in?

**Review by compnerd — DISMISSED** [review]

**Comment by hughbe** [comment]
@swift-ci please smoke test

File: tools/swift-demangle/swift-demangle.cpp
Comment (hughbe) [inline tools/swift-demangle/swift-demangle.cpp]: Sure - what I'm hoping for is to actually be able to get rid of this whole thing, and just import `<iostream>` for `std::getline`, so this might actually all go!

File: tools/swift-demangle/swift-demangle.cpp
Comment (hughbe) [inline tools/swift-demangle/swift-demangle.cpp]: Not sure. I've cleaned up the headers in the latest commit, but I haven't tested it on multiple platforms, so we'll see if it all compiles :)

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
Wonderful! looks like the tests are passing!

This was caused by the fact that POSIX `getline` includes a newline `\n` in the resultant string, but `std::getline` doesn't.

Fixed by appending a '\n' to the printing to `llvm::outs()`

**Comment by jrose-apple** [comment]
I guess we don't care about `<iostream>`'s static constructors this time?

**Comment by jrose-apple** [comment]
(That's the main reason [LLVM doesn't use it](http://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden).)

**Comment by jckarter** [comment]
We do. We should avoid importing `<iostream>`.

**Comment by hughbe** [comment]
Interesting, considering I actually got this code pattern from LLVM: https://github.com/llvm-mirror/llvm/blob/master/tools/llvm-cxxfilt/llvm-cxxfilt.cpp

@compnerd was the author there.

What's the alternative, id like to remove unistd imports and make this platform independent.

**Comment by jrose-apple** [comment]
I don't think it's so bad for a standalone tool like swift-demangle. If the constructors *weren't* static we'd still have to run them immediately just to read from stdin. (Admittedly swift-demangle doesn't always read from stdin.)

**Comment by jrose-apple** [comment]
We also do care about the streaming behavior (#4412).

**Comment by hughbe** [comment]
Okay, does the code look good in its current form? I'll defer to you both for a decision on the high level direction.

**Comment by jrose-apple** [comment]
I'm okay with it if @jckarter's convinced but since @compnerd started the review I'd let him sign off before merging.

**Comment by jckarter** [comment]
I guess it's fine for now. LLVM doesn't have a similar portable API?

**Comment by compnerd** [comment]
LGTM

**Pull request merged**
